yang/vyatta-protocols-frr-isis-v1.yang

```diff
@@ -0,0 +1,878 @@
+          				type uint32 {
+            					range "1..600";
+          				}
+          				units "seconds";
+        				configd:help "Holding time for level-1; interval will depend on multiplier.";
+        				description "Holding time for level-1; interval will depend on multiplier.";
+        			}
+
+        			leaf level-2 {
+          				type uint32 {
+            					range "1..600";
+          				}
+          				units "seconds";
+        				configd:help "Holding time for level-2; interval will depend on multiplier.";
+        				description "Holding time for level-2; interval will depend on multiplier.";
+        			}
+        		}
+
+        		container metric {
+	  			configd:help "Metric for this IS-IS circuit.";
+          			description "Metric for this IS-IS circuit.";
+          			leaf level-1 {
+            				type uint32 {
+              					range "0..16777215";
+           	 			}
+	    				configd:help "Level-1 metric for this IS-IS circuit.";
+            				description "Level-1 metric for this IS-IS circuit.";
+          			}
+
+          			leaf level-2 {
+            				type uint32 {
+              					range "0..16777215";
+            				}
+	    				configd:help "Level-2 metric for this IS-IS circuit.";
+            				description "Level-2 metric for this IS-IS circuit.";
+          			}
+        		}
+        		container priority {
+          			configd:help "Priority for Designated Router election."; 
+          			description "Priority for Designated Router election.";
+          			leaf level-1 {
+            				type uint8 {
+              					range "0..127";
+            				}
+	    				configd:help "Level-1 priority for this IS-IS circuit.";
+            				description "Level-1 priority for this IS-IS circuit.";
+          			}
+
+          			leaf level-2 {
+           				type uint8 {
+              					range "0..127";
+            				}
+	    				configd:help "Level-2 priority for this IS-IS circuit.";
+            				description "Level-2 priority for this IS-IS circuit.";
+          			}
+        		}
+        		leaf passive {
+          			type empty;
+	  			configd:help "Interface is in passive mode.";
+          			description "Interface is in passive mode.";
+        		}
+
+        		container password {
+          			presence "Present if a password is set for this IS interface.";
+	  			configd:help "Authentication password for an IS-IS area.";
+          			description "Authentication password for an IS-IS area.";
+          			uses isis-intf-password;
+        		}
+        		leaf network {
+          			type network-type;
+          			must "(. = \"point-to-point\") or (. = \"broadcast\")";
+	  			configd:help "Explicitly configured type of IS-IS circuit (broadcast or point-to-point)."; 
+          			description "Explicitly configured type of IS-IS circuit (broadcast or point-to-point).";
+        		}
+    		}
+  	}
+
+  	augment /if:interfaces/interfaces-dataplane:dataplane {
```

Comment:
We generally do interface augmentations as separate YANG files that import a grouping in the main file for the feature. We do this for modularity reasons so that not all interface types must be available on all builds.

Reply:
Thanks @jsouthworth , for the review. 
Hi @jsouthworth , 
We followed the "vyatta-protocols-frr-ospf-v1.yang" to perform the augmentations in the new file named "vyatta-protocols-frr-isis-v1.yang".
Since you are suggesting a different approach, can you please share a relevant file or example of it.

Reply:
OK, what is here should be sufficient then. Our guys didn't follow the convention when doing integration for the rest of the routing suite. It is highly unlike that an image would ever be built with out dataplane interfaces anyway. If support for other interface types are added in future release we'll need to follow the "tertiary model" convention for them.